File: src/dataset/c/pangram_Exercism_m3g4d1v3r/openAI/ChatGPT4_pangram_v3.c
#include "pangram.h"

bool is_pangram(const char *sentence) {
    if (sentence == NULL) return false;

    bool alphabet_seen[ALPHABET_SIZE] = { false };
    int unique_letters = 0;

    for (const char *ptr = sentence; *ptr != '\0'; ++ptr) {
        char chr = *ptr;
        if (chr >= 'a' && chr <= 'z') {
            chr -= 'a';
        } else if (chr >= 'A' && chr <= 'Z') {
            chr -= 'A';
        } else {
            continue;
        }

        if (!alphabet_seen[(int)chr]) {
            alphabet_seen[(int)chr] = true;
            unique_letters++;
            if (unique_letters == ALPHABET_SIZE) {
                return true;
            }
        }
    }

    return false;
}

File: src/dataset/typescript/pangram_Exercism-typescript-alemarr/openAI/ChatGPT4_pangram_v3.ts
const TOTAL_CHARS = 26;

export const isPangram = (sentence: string): boolean => {
  const seen = new Set<string>();
  for (let i = 0; i < sentence.length; i++) {
    const char = sentence[i].toLowerCase();
    if (char >= 'a' && char <= 'z') {
      seen.add(char);
      if (seen.size === TOTAL_CHARS) return true;
    }
  }
  return false;
};